presto-tests/src/test/java/com/facebook/presto/resourcemanager/TestDistributedClusterStatsResource.java

```diff
@@ -79,25 +111,77 @@ private String getResourceFilePath(String fileName)
         return this.getClass().getClassLoader().getResource(fileName).getPath();
     }
 
-    @Test(timeOut = 60_000, enabled = false)
+    @Test(timeOut = 220_000)
     public void testClusterStatsRedirectToResourceManager()
             throws Exception
     {
-        runToExecuting(client, coordinator, "SELECT avg(totalprice) from tpch.sf100.orders");
-
-        // Sleep to allow query to make some progress
-        sleep(SECONDS.toMillis(20));
-
-        ClusterStatsResource.ClusterStats clusterStats = getClusterStats(true);
-
+        waitUntilCoordinatorsDiscoveredHealthyInRM(SECONDS.toMillis(15));
+        runToFirstResult(client, coordinator2, "SELECT * from tpch.sf102.orders");
```

Comment:
nit maybe create an list of running queries and list of waiting queries as static array at class declaration? 
Then you can use size of array/list to check below as well

Reply:
We don't want to use a static array field for this because this is very specific to this test case. In this test we use "runToFirstResult" for the first 3 queries to keep those running, then we expect the 4th query to be queued since since it exceeds the limit set in resource group configuration. We may want to have other new tests where we may want to call runToFirstResult for 2 queries and perform runToQueued for the 3rd query and expect all these to be running. We could use @DataProvider but this could be a bit overkill to use for this test. wdyt?

Reply:
I think in the long run it will save if you think there are more scenarios that you will add. It is up to you in that regard.

Reply:
As we discussed, we will use @dataProvider  when we want to add more scenarios to the same test method. Since at this point, it seems like a overkill, I am not adding @dataProvider for this test.